homeassistant/components/ecobee/climate.py

```diff
@@ -330,6 +335,23 @@ async def async_update(self):
         if self.hvac_mode != HVAC_MODE_OFF:
             self._last_active_hvac_mode = self.hvac_mode
 
+        await self.throttled_validate_data()
+
+    @Throttle(VALIDATE_DATA_INTERVAL)
+    async def throttled_validate_data(self):
```

Comment:
This doesn't need to be a coroutine function since we don't await inside.

Reply:
I've moved the check into `async_setup_entry`. So `throttled_validate_data` goes away.